src/golang/cmd/server/handler/connect_integration.go

```diff
@@ -269,9 +267,20 @@ func ConnectIntegration(
 	}
 
 	if args.Service == integration.Conda {
-		if err = initializeConda(integrationObject.Id, integrationWriter, db); err != nil {
-			return http.StatusInternalServerError, errors.Wrap(err, "Unable to initialize Conda environments.")
-		}
+		go func() {
+			db, err = database.NewDatabase(db.Config())
```

Comment:
@likawind why do we need to create a new database object here instead of reusing the original `db` object?

Reply:
Reusing the object will cause connection error. I believe it's because the `Database` object is maintaining an underlying connection, which is terminated in go routine.